#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *
PACKAGE = "laser_filters"

gen = ParameterGenerator()

##### gen.add("name", paramtype, level, description, default, min, max)

gen.add("max_x", double_t, 0, "Maximum x coordinate of box", 1, -20, 20)
gen.add("max_y", double_t, 0, "Maximum y coordinate of box", 1, -20, 20)
gen.add("max_z", double_t, 0, "Maximum z coordinate of box", 1, -20, 20)
gen.add("min_x", double_t, 0, "Minimum x coordinate of box", -1, -20, 20)
gen.add("min_y", double_t, 0, "Minimum y coordinate of box", -1, -20, 20)
gen.add("min_z", double_t, 0, "Minimum z coordinate of box", -1, -20, 20)
gen.add("invert", bool_t, 0, "Invert", True)
gen.add("switch_", bool_t, 0, "Switches filter on or off", True)

exit(gen.generate(PACKAGE, "laser_filters", "BoxFilter"))